book/src/installer.md

```diff
@@ -0,0 +1,213 @@
+#### Fetch and run a pre-built installer
+
+With a well-known release URL, a pre-built binary can be obtained for supported
+platforms:
+
+```bash
+$ curl -o solana-install https://github.com/solana-labs/solana/releases/download/v1.2.3/solana-install-x86_64-apple-darwin
+$ chmod +x ./solana-install
+$ ./solana-install --help
+```
+
+Note: future enhancements should include building wrapper in the style of
+https://rustup.rs/ to make bootstrapping an install even easier.
+
+#### Build and run the installer from source
+
+If a pre-built binary is not available for a given platform, building the
+installer from source is always an option:
+```bash
+$ git clone https://github.com/solana-labs/solana.git
+$ cd solana/install
+$ cargo run -- --help
+```
+
+#### Deploy a new update to a cluster
+Given a solana release tarball (as created by `ci/publish-tarball.sh`) that has already been uploaded to a publicly accessible URL,
+the following commands will deploy the update:
+```bash
+$ solana-keygen -o update-manifest.json  # <-- only generated once, the public key is shared with users
+$ solana-install deploy http://example.com/path/to/solana-release.tar.bz2 update-manifest.json
+```
+
+Note: Supporting IPFS download URLs in the future would be attractive.
+
+#### Run a validator node that auto updates itself
+
+```bash
+$ solana-install init --pubkey 92DMonmBYXwEMHJ99c9ceRSpAmk9v6i3RdvDdXaVcrfj  # <-- pubkey is obtained from whoever is deploying the updates
+$ export PATH=~/.local/share/solana-install/bin:$PATH
+$ solana-keygen ...  # <-- runs the latest solana-keygen
+$ solana-install run solana-fullnode ...  # <-- runs a fullnode, restarting it as necesary when an update is applied
+```
+
+### Update Manifest on-chain program
+The Update Manifest program is used to advertise the deployment of new release tarballs
+on a solana cluster.  Each instance of this program describes a logical update
+channel for a given target triple (eg, `x86_64-apple-darwin`).  The public key
+of each program instance is well-known between the entity deploying new updates
+and users consuming those updates.
+
+The update tarball itself is hosted elsewhere, off-chain and can be fetched from
+the specified `download_url`.
+
+```rust,ignore
+use solana_sdk::signature::Signature;
+
+/// Information required to download and apply a given update
+pub struct UpdateManifest {
+  pub target: String,                 // Target triple (TARGET)
+  pub commit: String,                 // git sha1 of this update, must match the commit sha1 in the release tar.bz2
+  pub timestamp_secs: u64,            // When the release was deployed (seconds since UNIX EPOC)
+  pub download_url: String,           // Download URL to the release tar.bz2
+  pub download_signature: Signature,  // Signature of the release tar.bz2 file, verify with the Account public key
+}
+
+/// Userdata of an Update Manifest program Account.
+pub struct SignedUpdateManifest {
+  pub manifest: UpdateInfo,
```

Comment:
UpdateManifest?

Reply:
Sorry I can’t parse, need a couple more words please

Reply:
should this line read `pub manifest: UpdateManifest` ?

Reply:
Oh yep thanks